osh/cmd_eval.py

```diff
@@ -231,8 +231,19 @@ def PlusEquals(old_val, val):
                 raise AssertionError()  # parsing should prevent this
 
         elif case(value_e.BashAssoc):
-            # TODO: Could try to match bash, it will append to ${A[0]}
-            pass
+            if tag == value_e.Str:
+                e_die("Can't append string to array")
```

Comment:
```suggestion
                e_die("Can't append string to associative arrays")
```

Oh, the error message should be "associative arrays"...